rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/geo/TbGpsMultiGeofencingActionNode.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright © 2016-2024 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.rule.engine.geo;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.gson.JsonObject;
+import com.nimbusds.oauth2.sdk.util.CollectionUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.thingsboard.common.util.JacksonUtil;
+import org.thingsboard.rule.engine.api.RuleNode;
+import org.thingsboard.rule.engine.api.TbContext;
+import org.thingsboard.rule.engine.api.TbNodeConfiguration;
+import org.thingsboard.rule.engine.api.TbNodeException;
+import org.thingsboard.server.common.data.AttributeScope;
+import org.thingsboard.server.common.data.DataConstants;
+import org.thingsboard.server.common.data.EntityType;
+import org.thingsboard.server.common.data.id.AssetId;
+import org.thingsboard.server.common.data.id.DeviceId;
+import org.thingsboard.server.common.data.id.EntityId;
+import org.thingsboard.server.common.data.kv.AttributeKvEntry;
+import org.thingsboard.server.common.data.msg.TbMsgType;
+import org.thingsboard.server.common.data.plugin.ComponentType;
+import org.thingsboard.server.common.data.relation.*;
+import org.thingsboard.server.common.msg.TbMsg;
+import org.thingsboard.server.common.msg.TbMsgMetaData;
+
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+@Slf4j
+@RuleNode(type = ComponentType.ACTION,
+        name = "gps multi geofencing events",
+        configClazz = TbGpsMultiGeofencingActionNodeConfiguration.class,
+        relationTypes = {"Success", "Entered", "Left", "Inside", "Outside"},
+        nodeDescription = "Produces incoming messages using GPS based geofencing",
+        nodeDetails = "Extracts latitude and longitude parameters from incoming message and returns different events " +
+                "based on configuration parameters")
```

Comment:
Will be nice to have some additional details. The current description is poor and doesn't describe why this node has keyword "multi".

Reply:
@ShvaykaD The already existing geofence rule node works with only one perimeter that you define in the node configuration either in message itself. The new node works with many perimeters fetched by entity relations query. The node has keyword multi because instead of single perimeter, it works with many geofences or perimeters